test/e2e/snaps/test-snap-installed-metric.spec.js

```diff
@@ -0,0 +1,107 @@
+const { strict: assert } = require('assert');
+const {
+  defaultGanacheOptions,
+  withFixtures,
+  unlockWallet,
+  getEventPayloads,
+} = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+const { TEST_SNAPS_WEBSITE_URL } = require('./enums');
+
+async function mockSegment(mockServer) {
+  return [
+    await mockServer
+      .forPost('https://api.segment.io/v1/batch')
+      .withJsonBodyIncluding({
+        batch: [{ type: 'track', event: 'Snap Installed' }],
+      })
+      .thenCallback(() => {
+        return {
+          statusCode: 200,
+        };
+      }),
+  ];
+}
+
+describe('Snap Installed Event', function () {
+  it('successfully tracked when snap is installed', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withMetaMetricsController({
+            metaMetricsId: 'fake-metrics-id',
+            participateInMetaMetrics: true,
+          })
+          .build(),
+        defaultGanacheOptions,
+        title: this.test.title,
+        testSpecificMock: mockSegment,
+      },
+      async ({ driver, mockedEndpoint: mockedEndpoints }) => {
+        await driver.navigate();
+
+        await unlockWallet(driver);
+
+        // navigate to test snaps page and connect
+        await driver.openNewPage(TEST_SNAPS_WEBSITE_URL);
+        await driver.delay(1000);
+        const confirmButton = await driver.findElement('#connectDialogSnap');
+        await driver.scrollToElement(confirmButton);
+        await driver.delay(500);
+        await driver.clickElement('#connectDialogSnap');
+        await driver.delay(500);
+
+        // switch to metamask extension and click connect
+        const windowHandles = await driver.waitUntilXWindowHandles(
+          3,
+          1000,
+          10000,
+        );
+
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+
+        await driver.clickElement({
+          text: 'Connect',
+          tag: 'button',
+        });
+
+        await driver.waitForSelector({ text: 'Install' });
```

Comment:
Actually in retrospect... @DDDDDanica / @PeterYinusa did you say we should prefer testid versus text? If one doesn't exist is it possible we could add it for these tests?

Reply:
There is a testid for that specific element, `page-container-footer-next` which is used for notification button to proceed to the next screen for the 2-3 screens we show during a snap installation. I think the waitForSelector for the text to change is fine, but change the clickElement to the testid would be beneficial. Thoughts?

Reply:
@PeterYinusa or @DDDDDanica can you provide guidance here for both me and @tmashuang :)

Reply:
Personally i found if there're few confirmation screens and they all share `page-container-footer-next` id, it is going to cause flakiness at one stage. 
> I think the waitForSelector for the text to change is fine, but change the clickElement to the testid would be beneficial.
This sounds like a great idea, we confirm the button text is changed then click by its id... let's give it a try !!

@tmashuang if you could run the tests X times locally to test flakyness that would be awesome